MacrobenchmarkSample/ftl/README.md

```diff
@@ -0,0 +1,292 @@
+# Will launch a browser and complete an OAuth flow.
+firebase login
+
+# Choose a default GCP project
+gcloud config set project <your_project_id>
+```
+
+Update the `.firebaserc` file with your `project_id`.
+
+```json
+{
+  "projects": {
+    "default": "<your_project_id>"
+  }
+}
+```
+
+Now, deploy the Google Cloud Function.
+
+```bash
+# Will deploy the cloud functions for the selected GCP project.
+firebase deploy --only functions
+```
+
+##### Setting up environment variables
+
+The cloud function uses a couple of environment variables, to determine the `package_name` and the `device_configuration`s. After you `deploy` the cloud function for the first time, you will need to add these environment variables.
+
+To add the variables, go to the [Google Cloud Platform Console](https://console.cloud.google.com) and search for `Cloud Functions`. Then select a function called `completionHandler` and click on `Edit`.
+
+Now click on `Runtime, Build and Connection Settings`. Update the settings to the following:
+
+* Update `Memory allocated` to `4GiB`. 
+* Update `Timeout` to `500` seconds.
+* Add the following [environment variables](https://cloud.google.com/functions/docs/env-var).
+
+```bash
+# The Macrobenchmark target app package name
+# Example:
+package_name = "com.example.macrobenchmark.target"
+```
+
+```bash
+# The device configurations you are interested in tracking metrics for.
+# This should match the device that you are testing on in the GitHub workflow.
+# Example:
+device_configurations = ["flame-30-en-portrait"]
+```
+
+![Example setup](./images/cloud_function_settings.png)
+![Environment variables](./images/cloud_function_configuration.png)
+
+#### Invoking the Cloud Function
+
+The Google Cloud Function deployed should automatically get invoked _after_ a pull request is merged. You can track progress using the GitHub [Actions](https://docs.github.com/en/actions) tab.
+
+[Here](https://github.com/android/performance-samples/actions) is an example.
+
+![GitHub Actions](./images/github_actions.png)
+
+Once a Firebase Test Lab run is complete, the cloud function will be invoked. Under `Cloud Functions`, when you click on `Logs` you should see something like
+
+![Cloud Function Invocation](./images/cloud_function_invocation.png)
+
+
+### Google Cloud Monitoring
+
+All Macrobenchmark results are stored as [custom metrics](https://cloud.google.com/monitoring/custom-metrics) using Google Cloud Monitoring APIs. These metrics have a namespace `custom.googleapis.com` so you can use the [Metrics Query Editor](https://cloud.google.com/monitoring/mql/query-editor) to query and create dashboards and alerts to catch regressions.
+
+For more information on how you can use the MQL (query language) to create alerts, please refer to the [following documentation](https://cloud.google.com/monitoring/mql/alerts).
+
+#### Monitoring Dashboard
+
+![Monitoring Dashboard](./images/monitoring.png)
+
+#### Example Metrics
+
```

Comment:
was there supposed to be another image here?

Reply:
I added 2 images originally. The second one was kind of redundant so I got rid of it. 
Removed a title block to simplify this part.